src/UrlParams.ts

```diff
@@ -267,79 +300,131 @@ export const getUrlParams = (
 
   const fontScale = parseFloat(parser.getParam("fontScale") ?? "");
 
-  let intent = parser.getParam("intent");
-  if (!intent || !Object.values(UserIntent).includes(intent as UserIntent)) {
-    intent = UserIntent.Unknown;
-  }
-
-  // Check hideHeader for backwards compatibility. If header is set, hideHeader
-  // is ignored.
-  const header =
-    parser.getParam("header") ??
-    (parser.getFlagParam("hideHeader")
-      ? HeaderStyle.None
-      : HeaderStyle.Standard);
-
-  const sendNotificationType = ["ring", "notification"].includes(
-    parser.getParam("sendNotificationType") ?? "",
-  )
-    ? (parser.getParam("sendNotificationType") as RTCNotificationType)
-    : undefined;
   const widgetId = parser.getParam("widgetId");
   const parentUrl = parser.getParam("parentUrl");
   const isWidget = !!widgetId && !!parentUrl;
 
-  return {
+  /**
+   * The user's intent with respect to the call.
+   * e.g. if they clicked a Start Call button, this would be `start_call`.
+   * If it was a Join Call button, it would be `join_existing`.
+   * This is a platform specific default set of parameters, that allows to minize the configuration
+   * needed to start a call. And empowers the EC codebase to control the platform/intent behavior in
+   * a central place.
+   *
+   * In short: either provide url query parameters of UrlConfiguration or set the intent
+   * (or the global defaults will be used).
+   */
+  const intent = !isWidget
+    ? UserIntent.Unknown
+    : (parser.getEnumParam("intent", UserIntent) ?? UserIntent.Unknown);
+  // Here we only use constants and `platform` to determine the intent preset.
+  let intentPreset: UrlConfiguration;
+  switch (intent) {
+    case UserIntent.StartNewCall:
+      intentPreset = {
+        confineToRoom: true,
+        appPrompt: false,
+        preload: true,
+        header: platform === "desktop" ? HeaderStyle.None : HeaderStyle.AppBar,
+        showControls: true,
+        hideScreensharing: false,
+        allowIceFallback: true,
+        perParticipantE2EE: true,
+        controlledAudioDevices: platform === "desktop" ? false : true,
+        skipLobby: true,
+        returnToLobby: false,
+        sendNotificationType: "notification",
+      };
+      break;
+    case UserIntent.JoinExistingCall:
+      intentPreset = {
+        confineToRoom: true,
+        appPrompt: false,
+        preload: true,
+        header: platform === "desktop" ? HeaderStyle.None : HeaderStyle.AppBar,
+        showControls: true,
+        hideScreensharing: false,
+        allowIceFallback: true,
+        perParticipantE2EE: true,
+        controlledAudioDevices: platform === "desktop" ? false : true,
+        skipLobby: false,
+        returnToLobby: false,
+        sendNotificationType: undefined,
```

Comment:
This should be `"notification"`, because it's possible to start a call under this intent too if the previous call ends while you're on the lobby screen.

Reply:
And the logic for checking this would run in EC right?